bundles/org.eclipse.osgi/felix/src/org/apache/felix/resolver/ResolverImpl.java

```diff
@@ -91,24 +85,40 @@ static class ResolveSession implements Runnable
         private final ConcurrentMap<String, List<String>> m_usesCache = new ConcurrentHashMap<String, List<String>>();
         private ResolutionError m_currentError;
         volatile private CancellationException m_isCancelled = null;
+        private Logger logger;
+        private ToIntFunction<PermutationType> remaining = type -> {
```

Comment:
Both logger and remaining should be `final`

Reply:
Done